pennylane/operation.py

```diff
@@ -891,6 +891,15 @@ def __init__(self, *params, wires=None, do_queue=True, id=None):
                 f"{len(self._wires)} wires given, {self.num_wires} expected."
             )
 
+        # Check that the wires is not an empty list for operators that have
+        # `num_wires` as `AnyWires` or `AllWires`.
+        if self.num_wires in {AllWires, AnyWires}:
+            if not isinstance(self, (qml.Barrier, qml.Snapshot, qml.Hamiltonian)):
+                # Barrier, Snapshot: Applied to all wires if instantiated with wires = [].
+                # Hamiltonian: Possible to be empty with simplify().
+                if len(qml.wires.Wires(wires)) == 0:
+                    raise ValueError("require valid wires.")
```

Comment:
The error message is too cryptic. Can we give a bit more information?

Reply:
Sure. I made the following change to the error message.
```if len(qml.wires.Wires(wires)) == 0:
                    raise ValueError(
                        f"{self.name}: wrong number of wires. "
                        f"At least one wire has to be given."
                    )
```